tools/tool-testing/matcher.js

```diff
@@ -72,11 +72,13 @@ export default class Matcher {
 
     let timer = null;
     if (timeout) {
+      const failure = new TestFailure('match-timeout', {
+        run: this.run,
+        pattern: this.matchPattern,
+      });
```

Comment:
Creating the `TestFailure` outside the `setTimeout` callback is important here (and elsewhere) so that the correct stack trace is captured.